isomerpages/isomercms-frontend
# feat: bare version of the site dashboard

ℹ️ **Please review this PR by commits instead** ℹ️

## Problem



As part of identity phase 2, we are moving away from using GitHub for authentication and access management. However, these new features do not fit well into the existing Isomer CMS interface and workflow, as the existing screens are catered to changing content and settings on the actual site, rather than supporting processes that happen exclusively on the Isomer CMS.

## Solution



**Breaking Changes**




**Features**:

- The site layouts have been refactored and split into an edit and a view layout. The existing workspace layout is the SiteEditLayout (renamed from SiteViewLayout) whereas the new layout used by the identity phase 2 features is the new SiteViewLayout.
- The header used for both the SiteEditLayout is also preserved, whereas the header for the SiteViewLayout is a temporary placeholder that will be replaced by @alexanderleegs as part of the notifications feature.
- The existing ButtonLink component used in the site header has been abstracted out into a common component.
- 2 new images have been added: EmptyBoxPlainImage and SiteDashboardHumanImage
- A new common component is introduced, known as DisplayCard. This is adapted from the existing Card component but modelled as a Box. It also allows for more customisations to the DisplayCard, which is showcased in the Chromatic storybook.
- A new barebones site dashboard screen has been implemented. This contains the actual site dashboard but lacks the integration with the backend

**Caveats**:

- Some hooks have been added, namely useGetCollaboratorsStatistics, useGetReviewRequests and useGetSiteInfo. These hooks are subject to changes and may be removed in the future, especially with the collaborators and review requests features being introduced.
- The integration to the backend has not yet been implemented completely and may change rapidly in the future.
- The dashboard itself is not fully implemented yet, some additional modifications will be needed.

## Before & After Screenshots

**AFTER**:

<img width="1428" alt="Screenshot 2022-09-29 at 09 39 03" src="https://user-images.githubusercontent.com/27919917/192918959-f063bb98-118b-4e22-a0da-49cd3a71d549.png">

## Tests




1. Run `npm run dev` on this branch.
2. Navigate to any site and change the URL from "workspace" to "dashboard" (e.g. `/sites/zhongjun-test/dashboard`)
3. Verify that the new bare site dashboard loads.

## Deploy Notes




*None*
